RecoLocalTracker/SiPixelRecHits/interface/pixelCPEforGPU.h

```diff
@@ -0,0 +1,343 @@
+      DetParams const* __restrict__ l = m_detParams;
+      return l[i];
+    }
+    constexpr LayerGeometry const& __restrict__ layerGeometry() const { return *m_layerGeometry; }
+    constexpr AverageGeometry const& __restrict__ averageGeometry() const { return *m_averageGeometry; }
+
+    __device__ uint8_t layer(uint16_t id) const {
+      return __ldg(m_layerGeometry->layer + id / phase1PixelTopology::maxModuleStride);
+    };
+  };
+
+  // SOA (on device)
+  template <uint32_t N>
+  struct ClusParamsT {
+    uint32_t minRow[N];
+    uint32_t maxRow[N];
+    uint32_t minCol[N];
+    uint32_t maxCol[N];
+
+    int32_t Q_f_X[N];
+    int32_t Q_l_X[N];
+    int32_t Q_f_Y[N];
+    int32_t Q_l_Y[N];
+
+    int32_t charge[N];
+
+    float xpos[N];
+    float ypos[N];
+
+    float xerr[N];
+    float yerr[N];
+
+    int16_t xsize[N];  // clipped at 127 if negative is edge....
+    int16_t ysize[N];
+  };
+
+  constexpr int32_t MaxHitsInIter = gpuClustering::maxHitsInIter();
+  using ClusParams = ClusParamsT<MaxHitsInIter>;
+
+  constexpr inline void computeAnglesFromDet(
+      DetParams const& __restrict__ detParams, float const x, float const y, float& cotalpha, float& cotbeta) {
+    // x,y local position on det
+    auto gvx = x - detParams.x0;
+    auto gvy = y - detParams.y0;
+    auto gvz = -1.f / detParams.z0;
+    // normalization not required as only ratio used...
+    // calculate angles
+    cotalpha = gvx * gvz;
+    cotbeta = gvy * gvz;
+  }
+
+  constexpr inline float correction(int sizeM1,
+                                    int Q_f,                        //!< Charge in the first pixel.
+                                    int Q_l,                        //!< Charge in the last pixel.
+                                    uint16_t upper_edge_first_pix,  //!< As the name says.
+                                    uint16_t lower_edge_last_pix,   //!< As the name says.
+                                    float lorentz_shift,            //!< L-shift at half thickness
+                                    float theThickness,             //detector thickness
+                                    float cot_angle,                //!< cot of alpha_ or beta_
+                                    float pitch,                    //!< thePitchX or thePitchY
+                                    bool first_is_big,              //!< true if the first is big
+                                    bool last_is_big)               //!< true if the last is big
+  {
+    if (0 == sizeM1)  // size 1
+      return 0;
+
+    float w_eff = 0;
+    bool simple = true;
+    if (1 == sizeM1) {  // size 2
+      //--- Width of the clusters minus the edge (first and last) pixels.
+      //--- In the note, they are denoted x_F and x_L (and y_F and y_L)
+      // assert(lower_edge_last_pix >= upper_edge_first_pix);
+      auto W_inner = pitch * float(lower_edge_last_pix - upper_edge_first_pix);  // in cm
```

Comment:
these are adapted from https://github.com/cms-sw/cmssw/blob/add3363d786762a716ef7009b35643d05afd4329/CondFormats/SiPixelTransient/src/SiPixelUtils.cc#L42-L48

For consistency, I will update them in this PR only after they are updated there.

Reply:
ok, it's acceptable as is to me for now.

Reply:
perhaps one day we/someone checks if using https://clang.llvm.org/extra/clang-tidy/checks/readability-identifier-naming.html is worth it

Reply:
I've actually given it a try but - at least for data members - the automatic fixup is broken: it will rename the variables in the class definition, but not updated them in the rest of the code base :-(

OTOH even running it in warning-only mode, without automatic fixup, would avoid these slipping in.

Reply:
See https://github.com/cms-sw/cmssw/pull/32591 .

Reply:
> perhaps one day we/someone checks if using https://clang.llvm.org/extra/clang-tidy/checks/readability-identifier-naming.html is worth it

> OTOH even running it in warning-only mode, without automatic fixup, would avoid these slipping in.

Could you open an issue about this?